Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Fleet/Registration.hs

```diff
@@ -46,19 +65,86 @@ data FleetOwnerLoginReq = FleetOwnerLoginReq
   }
   deriving (Generic, Show, Eq, FromJSON, ToJSON, ToSchema)
 
+data FleetOwnerRegisterReq = FleetOwnerRegisterReq
+  { firstName :: Text,
+    lastName :: Text,
+    mobileNumber :: Text,
+    mobileCountryCode :: Text,
+    merchantId :: Text,
+    city :: City.City,
+    fleetType :: Maybe FOI.FleetType,
+    panNumber :: Maybe Text,
+    gstNumber :: Maybe Text,
+    panImageId1 :: Maybe Text,
+    panImageId2 :: Maybe Text
+  }
+  deriving (Generic, Show, Eq, FromJSON, ToJSON, ToSchema)
+
+newtype FleetOwnerRegisterRes = FleetOwnerRegisterRes
+  { personId :: Text
+  }
+  deriving (Generic, Show, Eq, FromJSON, ToJSON, ToSchema)
+
 newtype FleetOwnerVerifyRes = FleetOwnerVerifyRes
   { authToken :: Text
   }
   deriving (Generic, Show, Eq, FromJSON, ToJSON, ToSchema)
 
+fleetOwnerRegister :: FleetOwnerRegisterReq -> Flow FleetOwnerRegisterRes
+fleetOwnerRegister req = do
+  let merchantId = ShortId req.merchantId
+  mobileNumberHash <- getDbHash req.mobileNumber
+  deploymentVersion <- asks (.version)
+  merchant <-
+    QMerchant.findByShortId merchantId
+      >>= fromMaybeM (MerchantNotFound merchantId.getShortId)
+  merchantOpCityId <- CQMOC.getMerchantOpCityId Nothing merchant (Just req.city)
+  let personAuth = buildFleetOwnerAuthReq merchant.id req
+  person <-
+    QP.findByMobileNumberAndMerchantAndRole req.mobileCountryCode mobileNumberHash merchant.id DP.FLEET_OWNER
+      >>= maybe (createFleetOwnerDetails personAuth merchant.id merchantOpCityId True deploymentVersion.getDeploymentVersion req.fleetType req.gstNumber) return
+  fork "Creating Pan Info for Fleet Owner" $ do
+    createPanInfo person.id merchant.id merchantOpCityId req.panImageId1 req.panImageId2 req.panNumber
```

Comment:
How are you making sure, it is not duplicating?

Reply:
fixed.